Add zipkin tracer and standard http transport

paolochiodi opened the pull request:
The standard http transport is optional and is required in order to use
the zipking tracer because the current version conflict with seneca-mesh.

I still need to make USE_MESH and env variable, update package.json to point to a published package and possibly update the README, but I think this is still valuable to give @rjrodger a taste of the current solution

paolochiodi committed:
Add zipkin tracer and standard http transport

The standard http transport is optional and is required in order to use
the zipking tracer because the current version conflict with seneca-mesh

paolochiodi committed:
fix github link

rjrodger commented on fanout/fanout-service.js:
```diff
@@ -21,6 +16,12 @@ require('seneca')({
   })
 
   .ready(function(){
+    this.add('info:entry', function(msg,done){
```
don't think you need to wait for ready here, pretty sure you can just .add directly

paolochiodi replied:
Unfortunately this is needed by the transport-configuration-plugin to avoid circular references.

I've built that plugin to create clients for every other service, not only the ones really used (this basically replicate the same behaviour of seneca mesh).

The problem though is that this service will both have a client with `pin:'info:entry'` and and action for `info:entry`. If the client is added last its info:entry will overwrite the local `info:entry` resulting in the service receiving the message from the transport and then resending it to index-service. index-service itself will have the same problem and will re-send again the message resulting in an infinite loop.

If the local action is added after the plugin it will prevail and when seneca will receive info:entry from the transport will follow the intended path.

I tried alternative approaches, but couldn't find one that broke the infinite loop (without major changes to ramanujan or the transport-configuration-plugin)

rjrodger commented on index/index-service.js:
```diff
@@ -22,5 +17,10 @@ require('seneca')({
   })
 
   .ready(function(){
+    this.add('info:entry', function(msg,done){
```
same again, not sure ready needed

paolochiodi replied:
see above

rjrodger commented on transport-config/transport-config.js:
```diff
@@ -0,0 +1,142 @@
+
```
awesome! you know this is worth making into a new standalone plugin :)
let's put that on the todo list

paolochiodi replied:
I'll add to trello :)

rjrodger reviewed (commented):
using this new github review feature!

paolochiodi committed:
add MESH env variable to control transport configuration

paolochiodi committed:
remove debug code

paolochiodi committed:
update README

paolochiodi commented:
I've updated the PR with the configuration for toggling mesh and http transport and updated readme. 

It still uses the github version of seneca-zipkin-tracer, but I think it's ok to merge.
@rjrodger what do you think?

paolochiodi commented:
@rjrodger I think you already merged this in a feature branch, should we close the PR?

paolochiodi commented:
closing because already present in the https://github.com/senecajs/ramanujan/tree/zipkin branch
